ui/pages/Applications.tsx

```diff
@@ -1,50 +1,132 @@
+import { Checkbox } from "@material-ui/core";
 import * as React from "react";
 import styled from "styled-components";
 import Button from "../components/Button";
 import DataTable from "../components/DataTable";
 import Flex from "../components/Flex";
+import Icon, { IconType } from "../components/Icon";
 import Link from "../components/Link";
 import Page, { TitleBar } from "../components/Page";
 import useApplications from "../hooks/applications";
 import { Application } from "../lib/api/applications/applications.pb";
 import { PageRoute } from "../lib/types";
 import { formatURL } from "../lib/utils";
-
 type Props = {
   className?: string;
 };
 
 function Applications({ className }: Props) {
   const [applications, setApplications] = React.useState<Application[]>([]);
+  const [sort, setSort] = React.useState("Name");
+  const [reverseSort, setReverseSort] = React.useState(false);
   const { listApplications, loading } = useApplications();
 
   React.useEffect(() => {
     listApplications().then((res) => setApplications(res as Application[]));
   }, []);
 
+  type labelProps = { label: string };
+  function SortableLabel({ label }: labelProps) {
+    return (
+      <Flex align>
+        <Button
+          onClick={() => {
+            setSort(label);
+            setReverseSort(false);
+          }}
+          className={`lowercase table ${sort === label && "bold"}`}
+        >
+          <p>{label}</p>
+        </Button>
+        {sort === label && (
+          <Button
+            onClick={() =>
+              reverseSort ? setReverseSort(false) : setReverseSort(true)
+            }
+            className="table"
+          >
+            <Icon
+              type={IconType.ArrowDownward}
+              size="small"
+              className={reverseSort ? "upward" : "downward"}
+            />
+          </Button>
+        )}
+      </Flex>
+    );
+  }
+
   return (
     <Page loading={loading} className={className}>
       <Flex align between>
         <TitleBar>
-          <h2>Applications</h2>
+          <h2>Installed GitOps Applications</h2>
         </TitleBar>
-        <Link to={PageRoute.ApplicationAdd} className="title-bar-button">
-          <Button variant="contained" color="primary" type="button">
-            Add Application
+      </Flex>
+      <Flex className="application-actions">
```

Comment:
How about splitting this into an `<ActionBar />` component? I feel like we will use this pattern a lot in the future.

I can imagine there will be lots of places where we want a bunch of buttons at the top of a page.

Reply:
Should I split the buttons into separate components and pass those down, or have them living in `<ActionBar />` somehow?

Reply:
I would say make them `children` of the `<ActionBar />`